homeassistant/components/alexa/entities.py

```diff
@@ -618,9 +618,10 @@ def interfaces(self):
             inputs = AlexaEqualizerController.get_valid_inputs(
                 self.entity.attributes.get(media_player.const.ATTR_SOUND_MODE_LIST, [])
             )
-            if len(inputs) > 0 and self.entity_conf.get(CONF_PLATFORM) not in [
-                "denonavr"
-            ]:
+            # AlexaEqualizerController is disabled for denonavr
+            # since it blocks alexa from discovering any devices.
+            domain = entity_sources(self.hass)[self.entity_id]['domain']
+            if len(inputs) > 0 and domain not in ["denonavr"]:
```

Comment:
The denon check should be moved to the parent if-statement. We don't need to even get the valid inputs if we are never going to return it.

Reply:
done